js/apps/system/_admin/aardvark/APP/react/src/views/views/Components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from "react";
+import AccordionItem from "./AccordionItem";
+import { IAccordionItemViewArgs, IAccordionViewArgs } from "./types";
+import "./styles.css";
+
+const AccordionView = ({
+  accordionConfig = [],
+  allowMultipleOpen
+}: IAccordionViewArgs) => {
+  const [activeIndices, setActiveIndices] = useState<Array<number>>([]);
+
+  useEffect(() => {
+    const defaultActiveTab = accordionConfig.filter(
+      (config) => config.defaultActive
+    );
+    if (defaultActiveTab.length) {
+      setActiveIndices([defaultActiveTab[0].index]);
+    }
+  }, []);
+
+  const handleClick = (index: number) => {
+    const doesIndexExist = !!activeIndices.filter((val) => val === index)
+      .length;
+
+    if (allowMultipleOpen) {
+      if (doesIndexExist) {
+        const updatedIndices = activeIndices.filter((val) => val !== index);
+        setActiveIndices([...updatedIndices]);
+      } else {
+        setActiveIndices([...activeIndices, index]);
+      }
+    } else {
+      if (doesIndexExist) {
+        setActiveIndices([]);
+      } else {
+        setActiveIndices([index]);
+      }
+    }
+  };
+
+  if (!accordionConfig.length || !accordionConfig) {
+    return null;
+  }
+  return (
+    <div className="ui styled accordion">
+      {accordionConfig.map((config: IAccordionItemViewArgs) => {
+        return (
+          <AccordionItem
+            {...config}
+            allowMultipleOpen={allowMultipleOpen}
+            key={config.index}
+            active={activeIndices.includes(config.index)}
+            onClick={handleClick}
```

Comment:
Can we rename this prop to something other than `onClick`, as this could be interpreted to be a click handler for this div instead of a prop that is transparently passed down.

Reply:
Good idea. Resolved.